### DISCUSSION TITLE
Angular: Issue trying to import katex

### DISCUSSION THREAD

**Discussion by ChavaMS** [KaTeX/KaTeX#3858] [Q&A]
Hi everyone! 

I'm trying to import Katex on an Angular project and I'm facing this error: 

`node_modules/@types/katex/index"' has no default export.`

This is how I imported the module:

![image](https://github.com/KaTeX/KaTeX/assets/51062516/3c13962e-5a56-4770-9427-99d92a4d8009)

I'm using: 
-Angular 16
-Katex 0.16.8

What am I doing wrong?

**Comment by edemaine** [comment]
I'm guessing this is a type error, and related to #3574 (and #3531). Presumably everything runs, it's "just" type checking that's broken.

**Reply by ChavaMS** [reply]
No, it doesn't run, the Angular compiler throws an error and stops working. 😢 

Taking a deeper look at the repository I found this PR that might help with the issue: https://github.com/KaTeX/KaTeX/pull/3846 but isn't approved yet. I think it is an important issue because it affects every typescript implementation.

**Comment by joshuawwright** [comment]
@edemaine @ChavaMS I have the same problem. I can kind of work around the issue by doing this:

`import { ElementRef, Injectable } from '@angular/core';
import { KatexOptions } from 'katex';
import { default as katex } from 'katex';

@Injectable()
export class KatexService {

    render(equation: string, element: ElementRef, options?: KatexOptions) {
        return katex.render(equation, element.nativeElement, options);
    }
}

**Comment by joshuawwright** [comment]
@ChavaMS It looks like this library managed to get it to work:

https://github.com/garciparedes/ng-katex/blob/master/projects/ng-katex/src/lib/ng-katex.service.ts

I am using NX workspace with Angular, so maybe the tsconfig/executors are playing a role here.

**Reply by joshuawwright** [reply]
The reason it works for them is because they don't have @types/katex as a dev dependency. Obviously, not an ideal solution.

**Comment by mittal2707** [comment]
@joshuawwright I saw your workaround and tried to implement the same, but it seems to not work for me. Here is my code : 
![image](https://github.com/user-attachments/assets/f988d869-f0cc-4d62-a511-396a05600f3a)
This is the error on default : Module '"katex"' has no exported member 'default'. Am I missing something in maybe tsconfig?

**Reply by mittal2707** [reply]
I am using 
"katex": "^0.16.11"
"@types/katex": "^0.16.7",

**Reply by mittal2707** [reply]
Cc : @edemaine

**Reply by mittal2707** [reply]
```ts
import { ElementRef, Injectable } from '@angular/core';
import { KatexOptions, render } from 'katex';

@Injectable()
export class KatexService {
  constructor() {}

  public render(equation: string, element: ElementRef, options?: KatexOptions) {
    return render(equation, element.nativeElement, options);
  }
}
```

this is the original code, which is not working and getting this error on angular build :  Error: export 'render' (imported as 'render') was not found in 'katex' (possible exports: default)
